lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -888,485 +679,90 @@ export default function getAPIImplementation(): APIInterface {
+                        u.loginMethods.some((lm) => lm.recipeId === "emailpassword" && lm.hasSameEmailAs(email))
+                    )
+                ) {
+                    return {
+                        status: "EMAIL_ALREADY_EXISTS_ERROR",
+                    };
                 }
-            };
-
-            const linkAccountsForFactorSetup = async (
-                sessionUser: User,
-                recipeUserId: RecipeUserId,
-                userContext: UserContext
-            ) => {
-                if (!sessionUser.isPrimaryUser) {
-                    const createPrimaryRes = await AccountLinking.getInstance().recipeInterfaceImpl.createPrimaryUser({
-                        recipeUserId: new RecipeUserId(sessionUser.id),
-                        userContext,
-                    });
-                    if (createPrimaryRes.status === "RECIPE_USER_ID_ALREADY_LINKED_WITH_PRIMARY_USER_ID_ERROR") {
-                        // Session user is linked to another primary user, which means the session is revoked as well
 
-                        // We cannot recurse here because when the session user if fetched again,
-                        // it will be a primary user and we will end up trying factor setup with that user
-                        // Also this session would have been revoked and we won't be able to catch it again
+                return {
+                    status: "SIGN_UP_NOT_ALLOWED",
+                    reason:
+                        "Cannot sign up due to security reasons. Please try logging in, use a different login method or contact support. (ERR_CODE_007)",
+                };
+            }
 
-                        throw new SessionError({
-                            type: SessionError.UNAUTHORISED,
-                            message: "Session may be revoked",
-                        });
-                    } else if (
-                        createPrimaryRes.status === "ACCOUNT_INFO_ALREADY_ASSOCIATED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR"
-                    ) {
-                        // We had determined during validation that the linking is going to pass, but now it has failed
-                        // due to another parallel request. So we recurse again from the validation phase and return
-                        // appropriate error from there
-                        throw new RecurseError();
-                    }
-                }
+            const overwriteSessionDuringSignInUp = SessionRecipe.getInstanceOrThrowError().config
+                .overwriteSessionDuringSignInUp;
 
-                const linkRes = await AccountLinking.getInstance().recipeInterfaceImpl.linkAccounts({
-                    recipeUserId: recipeUserId,
-                    primaryUserId: sessionUser.id,
+            let signUpResponse;
+            if (session === undefined || overwriteSessionDuringSignInUp) {
+                signUpResponse = await options.recipeImplementation.signUp({
```

Comment:
add a comment here saying that we have this and not call createNewRecipeUser, (even though semantically for this API we could) because users use signUp override to do things post sign up and not createNewRecipeUser override